### ISSUE TITLE
Reconnect Problems with 0.6.0

### ISSUE THREAD

**Issue by codmpm** [codmpm/node-red-contrib-loxone#14]
As already discussed in #9 and #11, there is a problem on reconnect from the node. Reproduceable when updating the Miniserver software from loxconfig.

Currently testing why it faults...

cc @mabunixda, @alladdin

**Comment by codmpm** [comment]
I saw that I've added a `client.abort()` on `close` which seems to be the problem. Naturally I thought, if the connection is closed the client should abort. But this event also get's fired when the miniserver disconnects from his side and so a reconnect is not tried.

Sadly node-lox-ws-api crashes on disconnect without `client.abort()` with the following message with node-lox-ws-api 0.4.2: 

```
0|node-red | 26 Oct 22:38:24 - [info] [loxone-miniserver:79307096.f7614] Miniserver connected (nodelox.fritz.box:80) using Token-Enc
0|node-red | 26 Oct 22:38:25 - [info] [loxone-miniserver:79307096.f7614] got structure file 2017-10-26 22:38:13
0|node-red | 26 Oct 22:39:48 - [info] [loxone-miniserver:79307096.f7614] connection closed
0|node-red | 26 Oct 22:39:48 - [info] [loxone-miniserver:79307096.f7614] connection closed
0|node-red | SyntaxError: Unexpected token < in JSON at position 0
0|node-red |     at Object.parse (native)
0|node-red |     at IncomingMessage.res.on (/home/pm/node-red-contrib-loxone/node_modules/node-lox-ws-api/lib/API.js:276:29)
0|node-red |     at emitOne (events.js:96:13)
0|node-red |     at IncomingMessage.emit (events.js:188:7)
0|node-red |     at IncomingMessage.Readable.read (_stream_readable.js:381:10)
0|node-red |     at flow (_stream_readable.js:761:34)
0|node-red |     at resume_ (_stream_readable.js:743:3)
0|node-red |     at _combinedTickCallback (internal/process/next_tick.js:80:11)
0|node-red |     at process._tickDomainCallback (internal/process/next_tick.js:128:9)
0|node-red | 26 Oct 22:39:50 - [red] Uncaught Exception:
0|node-red | 26 Oct 22:39:50 - SyntaxError: Unexpected token < in JSON at position 0
0|node-red |     at Object.parse (native)
0|node-red |     at IncomingMessage.res.on (/home/pm/node-red-contrib-loxone/node_modules/node-lox-ws-api/lib/API.js:276:29)
0|node-red |     at emitOne (events.js:96:13)
0|node-red |     at IncomingMessage.emit (events.js:188:7)
0|node-red |     at IncomingMessage.Readable.read (_stream_readable.js:381:10)
0|node-red |     at flow (_stream_readable.js:761:34)
0|node-red |     at resume_ (_stream_readable.js:743:3)
0|node-red |     at _combinedTickCallback (internal/process/next_tick.js:80:11)
0|node-red |     at process._tickDomainCallback (internal/process/next_tick.js:128:9)```

**Comment by codmpm** [comment]
Fixed in node-lox-ws-api `0.4.3` thanks to @alladdin. It is already working in my testing environment, but I have to fix some connection handling problems...
I think I'll release a new version of node-red-contrib-loxone this evening.

```
0|node-red | 27 Oct 09:01:21 - [info] [loxone-miniserver:79307096.f7614] Miniserver connected (nodelox.fritz.box:80) using Token-Enc
0|node-red | 27 Oct 09:01:21 - [info] [loxone-miniserver:79307096.f7614] got structure file 2017-10-27 09:01:16
0|node-red | 27 Oct 09:03:21 - [info] [loxone-miniserver:79307096.f7614] keepalive (1ms)
0|node-red | 27 Oct 09:04:09 - [info] [loxone-miniserver:79307096.f7614] connection closed
0|node-red | 27 Oct 09:04:11 - [info] [loxone-miniserver:79307096.f7614] connection closed
0|node-red | 27 Oct 09:04:13 - [info] [loxone-miniserver:79307096.f7614] Miniserver connected (nodelox.fritz.box:80) using Token-Enc
0|node-red | 27 Oct 09:04:14 - [info] [loxone-miniserver:79307096.f7614] got structure file 2017-10-27 09:04:08
```

**Comment by codmpm** [comment]
fixed in 0.6.1

I've removed the "activate connection" checkbox for now, as it's change could only be tracked on a full deploy (not modified flows/nodes) so the connection to the miniserver could not be closed accordingly...
